src/psyclone/psyGen.py

```diff
@@ -4465,6 +4663,227 @@ def _number_handler(self, node, parent):
+    :raises NotImplementedError: Provided parameters are not supported yet.
+    :raises TypeError: Provided parameters have invalid error type.
+    :raises ValueError: Provided parameters contain invalid values.
+    '''
+
+    # Tuple with the valid values for the access attribute.
+    valid_access_types = ('local', 'external', 'read_arg', 'write_arg',
+                          'readwrite_arg')
+
+    def __init__(self, name, datatype=None, shape=[], access=None):
+
+        if datatype not in ('real', 'integer'):
+            raise NotImplementedError("Symbol can only be initialized with "
+                                      "'real' or 'integer' datatypes.")
+
+        if not isinstance(shape, list):
+            raise TypeError("Symbol shape attribute must be a list.")
+
+        if False in [isinstance(x, (type(None), int)) for x in shape]:
+            raise TypeError("Symbol shape list elements can only be "
+                            "'integer' or 'None'.")
+
+        if access not in Symbol.valid_access_types:
+            raise ValueError("Symbol access attribute can only be: "
+                             "'local', 'external', 'read_arg', "
+                             "'write_arg' or 'readwrite_arg'.")
+
+        self._name = name
+        self._datatype = datatype
+        self._shape = shape
+        self._access = access
+
+    @property
+    def name(self):
+        '''
+        :return: Name of the Symbol.
+        :rtype: string
+        '''
+        return self._name
+
+    @property
+    def datatype(self):
+        '''
+        :return: Datatype of the Symbol.
+        :rtype: string
+        '''
+        return self._datatype
+
+    @property
+    def shape(self):
+        '''
+        :return: Shape of the symbol in row-major order (leftmost \
+                 index is contiguous in memory). Each entry represents \
+                 an array dimension. If not None then it holds the \
+                 extent of that dimension. If it is an empy list it \
+                 represents an scalar.
+        :rtype: list
+        '''
+        return self._shape
+
+    @property
+    def access(self):
+        '''
+        :return: Whether the variable is 'local', 'external' or an argument. In
+                 case it is argument it specifies whether it is a 'read_arg',
+                 'write_arg' or 'readwrite_arg' argument.
+        :rtype: str
+        '''
+        return self._access
+
+    @access.setter
+    def access(self, new_access):
+        '''
+        :param str new_access: New value of the access attribute, can be:
+                               'local', 'external', 'read_arg', 'write_arg'
+                               or 'readwrite_arg'.
+        :raises ValueError: New access parameter has an invalid value.
+        '''
+        if new_access not in Symbol.valid_access_types:
+            raise ValueError("Symbol access attribute can only be: "
```

Comment:
Please use `Symbol.valid_access_types` in the error message for future proofing.

Reply:
Done